### ISSUE TITLE
Input: move duplicate code to utils

### ISSUE THREAD

**Issue by benelan** [Esri/calcite-components#5171]
### Description

As we split Input into multiple components by type (Input Number, Input Text, Input Time, etc.) we are starting to get a lot of duplicate code. We should move the duplicate code to a new util: `src/utils/input.ts`.

### Proposed Advantages

It will make bugs easier to fix in the future when everything is in one place.

### Which Component

- `calcite-input`
- `calcite-input-text`
- `calcite-input-number`
- `calcite-input-time` (once @eriklharper is done)

Any new `calcite-input-` components we create should use the utils.

**Comment by eriklharper** [comment]
Original comment that sparked this issue for context: https://github.com/Esri/calcite-components/pull/5149#discussion_r947323426

**Comment by benelan** [comment]
Moving this lower priority issue to the next milestone since I grabbed some additional validation issues for this sprint. cc @jcfranco

**Comment by geospatialem** [comment]
Moving to November. 

Additional team discussions are needed if the inheritance proposal above should be added, or we should plan another strategy.

**Comment by isaacbraun** [comment]
Closing as duplicate of #8126, as a by-product of that effort is moving duplicate/shared code to a util.
